File: HomeworkManager/HomeworkManager.Model/CustomEntities/User/UserModel.cs
using HomeworkManager.Model.CustomEntities.Role;

namespace HomeworkManager.Model.CustomEntities.User;

public class UserModel
{
    public required Guid UserId { get; set; }
    public required string FullName { get; set; }
    public required string Username { get; set; }
    public required string Email { get; set; }
    public required bool EmailConfirmed { get; set; }
    public ICollection<RoleModel> Roles { get; set; } = new HashSet<RoleModel>();
}

File: HomeworkManager.App/app/src/main/java/hu/bme/aut/android/homeworkmanagerapp/domain/model/user/UserModel.kt
package hu.bme.aut.android.homeworkmanagerapp.domain.model.user

data class UserModel(
    val userId: String,
    val fullName: String,
    val username: String,
    val email: String,
    val emailConfirmed: Boolean,
    val roles: List<RoleModel>
)